data/data_loader.py

```diff
@@ -175,161 +151,37 @@ def __sort_and_precalc_image_ratings(self) -> tuple[float, list[float]]:
-                
-            except Exception as e:
-                logging.error(f"{Fore.LIGHTRED_EX} *** Error opening {Fore.LIGHTYELLOW_EX}{pathname}{Fore.LIGHTRED_EX} to get metadata. File may be corrupt and will be skipped.{Style.RESET_ALL}")
-                logging.error(f" *** exception: {e}")
-                pass
 
         if not self.has_scanned:
             self.has_scanned = True
-            if len(undersized_images) > 0:
-                underized_log_path = os.path.join(self.log_folder, "undersized_images.txt")
-                logging.warning(f"{Fore.LIGHTRED_EX} ** Some images are smaller than the target size, consider using larger images{Style.RESET_ALL}")
-                logging.warning(f"{Fore.LIGHTRED_EX} ** Check {underized_log_path} for more information.{Style.RESET_ALL}")
-                with open(underized_log_path, "w") as undersized_images_file:
-                    undersized_images_file.write(f" The following images are smaller than the target size, consider removing or sourcing a larger copy:")
-                    for undersized_image in undersized_images:
-                        undersized_images_file.write(f"{undersized_image}\n")
-        
-        print (f" * DLMA: {len(decorated_image_train_items)} images loaded from {len(image_paths)} files")
-
-        return decorated_image_train_items
+            
+            logging.info(" Preloading images...")
+            
+            items, events = resolver.resolve(self.data_root, self.aspects, flip_p=flip_p, seed=self.seed)
+            image_paths = set(map(lambda item: item.pathname, items))
+
+            print (f" * DLMA: {len(items)} images loaded from {len(image_paths)} files")
+            
+            self.prepared_train_data = items
+            random.Random(self.seed).shuffle(self.prepared_train_data)
+            self.__report_undersized_images(events)
+    
+    def __report_undersized_images(self, events: list[resolver.Event]):
+        events = [event for event in events if isinstance(event, resolver.UndersizedImageEvent)]
+
+        if len(events) > 0:
+            underized_log_path = os.path.join(self.log_folder, "undersized_images.txt")
+            logging.warning(f"{Fore.LIGHTRED_EX} ** Some images are smaller than the target size, consider using larger images{Style.RESET_ALL}")
+            logging.warning(f"{Fore.LIGHTRED_EX} ** Check {underized_log_path} for more information.{Style.RESET_ALL}")
+            with open(underized_log_path, "w") as undersized_images_file:
+                undersized_images_file.write(f" The following images are smaller than the target size, consider removing or sourcing a larger copy:")
+                for event in events:
+                    message = f" *** {event.image_path} with size: {event.image_size} is smaller than target size: {event.target_size}, consider using larger images"
+                    undersized_images_file.write(message)
```

Comment:
I put this stuff here because I thought it helped `__prescan_images` read a bit nicer. Totally doesn't have to go here though.